src/getstream/updateBetterChannelMembers.js

```diff
@@ -0,0 +1,176 @@
+const _ = require('lodash');
+const {Sequelize} = require('sequelize');
+const {sequelize} = require('../databases/models');
+const BetterSocialConstantListUtils = require('../utils/constantList/utils');
+
+const SEPARATOR = ', ';
+
+const getAllChatAnonimityUserInfo = async (channelId, userIds = []) => {
+  if (!channelId) throw new Error('Channel ID is required');
+
+  if (userIds?.length === 0) return [];
+  const query = `
+    SELECT
+        users.user_id,
+        users.username,
+        users.is_anonymous,
+        chat_anon_user_info.*
+    FROM users
+    LEFT JOIN chat_anon_user_info
+        ON users.user_id = CAST(chat_anon_user_info.my_anon_user_id as varchar)
+        AND chat_anon_user_info.channel_id = :channelId
+    WHERE users.user_id IN (:userIds)`;
+
+  try {
+    const result = await sequelize.query(query, {
+      replacements: {
+        channelId,
+        userIds
+      },
+      type: Sequelize.QueryTypes.SELECT
+    });
+
+    return result;
+  } catch (e) {
+    console.error('Failed to get chat anonimity user info', e);
+    throw e;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling in this function could be improved. Instead of just logging the error and rethrowing it, consider using a custom error class to provide more context about what went wrong. This can make debugging easier if an error occurs.

```diff
-    console.error('Failed to get chat anonimity user info', e);
-    throw e;
+    throw new DatabaseError('Failed to get chat anonimity user info', e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->